maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -332,10 +332,10 @@ void initialize(CliRequest cliRequest) throws ExitException {
         for (String arg : cliRequest.args) {
             if (isAltFile) {
                 // this is the argument following -f/--file
-                Path path = Paths.get(arg);
+                Path path = topDirectory.resolve(arg);
```

Comment:
That's really weird.  That would make file argument order dependent as the `topDirectory` is modified inside the loop.

Reply:
Even if this was not the case, the goal of the loop is to compute the `topDirectory`.
However, introducing a `workingDirectory` variable and using it to resolve the `path` could make sense.

Reply:
I still don't understand why would this loop enter 2 or more times? (the user specifies several -f parameters?)

Reply:
but even then, there is a break (or throws) so first entrance will either break or throw

Reply:
Missed the `break` ...